ondevice_learning.qmd

```diff
@@ -1,61 +1,630 @@
+
+Furthermore, the energy and environmental costs associated with running large server farms are also diminished. Data centers are known to consume vast amounts of energy, contributing to greenhouse gas emissions. By reducing the need for extensive cloud-based infrastructure, on-device learning plays a part in mitigating the environmental impact of data processing [@wu2022sustainable].
+
+Specifically for endpoint applications, on-device learning minimizes the number of network API calls needed to run inference through a cloud provider. For applications with millions of users, the cumulative costs associated with bandwidth and API calls can quickly escalate. In contrast, performing training and inferences locally is considerably more efficient and cost-effective. On-device learning has been shown to reduce training memory requirements, drastically improve memory efficiency, and reduce up to 20% in per-iteration latency under the state-of-the-art optimizations [@dhar2021survey].
+
+Another key benefit of on-device learning is the potential for Internet of Things devices to continuously adapt their machine-learning model to new data for continuous, lifelong learning. On-device models can quickly become outdated as user behavior, data patterns, and preferences change. Continuous learning enables the model to efficiently adapt to new data and improvements and maintain high model performance over time.
+
+### Limitations
+
+While traditional cloud-based machine learning systems have access to nearly endless computing resources, on-device learning is often restricted by the limitations in computational and storage power of the edge device that the model is trained on. By definition, an edge device is a device with restrained computing, memory, and energy resources, that cannot be easily increased or decreased. Thus, the reliance on edge devices can restrict the complexity, efficiency, and size of on-device machine learning models. 
+
+#### Compute resources
+
+Traditional cloud-based machine learning systems utilize large servers with multiple high-end GPUs or TPUs that provide nearly endless computational power and memory. For example, services like AWS EC2 allow configuring clusters of GPU instances for massively parallel training. 
+
+In contrast, on-device learning is restricted by the hardware limitations of the edge device it runs on. Edge devices refer to endpoints like smartphones, embedded electronics, and IoT devices. By definition, these devices have highly restrained computing, memory, and energy resources compared to the cloud. 
+
+For example, a typical smartphone or Raspberry Pi may only have a few CPU cores, a few GB of RAM, and a small battery. These resources are fixed and can't easily be increased on demand like scaling cloud infrastructure. This reliance on edge devices directly restricts the complexity, efficiency, and size of models that can be deployed for on-device training:
```

Comment:
I would also have a sentence mentioning tinyML devices, since these are often even more constrained than the typical edge device, for example, the Arduino Nano BLE Sense: https://store-usa.arduino.cc/products/arduino-nano-33-ble-sense